pkg/controller/routes.go

```diff
@@ -28,13 +28,14 @@ var (
 	})
 )
 
-func routes(router *router.Router, registryTransport http.RoundTripper) {
+func routes(router *routeUtils.Router, registryTransport http.RoundTripper) {
 	router.OnErrorHandler = appdefinition.OnError
 
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.AssignNamespace)
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.PullAppImage(registryTransport))
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.ParseAppImage)
 	router.HandleFunc(&v1.AppInstance{}, tls.ProvisionCerts) // Provision TLS certificates for port bindings with user-defined (valid) domains
+	router.HandleFunc(&v1.AppInstance{}, appdefinition.FilterLabelsAndAnnotationsConfig(routeUtils.HandlerFunc(namespace.AddNamespace)).Handle)
```

Comment:
nit: this makes it obvious that we are adding a middleware. Feel free to ignore if you disagree.
```suggestion
	router.Type(&v1.AppInstance{}).Middleware(appdefinition.FilterLabelsAndAnnotationsConfig).HandlerFunc(routeUtils.HandlerFunc(namespace.AddNamespace))
```

Reply:
Yeah it is fine i can change it.

Originally I wrote it in this way but change it since the test code has to be written in this way in order to test.(the test framework doesn't support middleware so it is easy to miss out).

I guess we just need to be careful to include middleware in testing code.